scripts/busybox-ping-test/create_test_cases.py

```diff
@@ -60,29 +65,45 @@ def create_security_groups(port):
        security_groups["security_group"] = sginfo
        print("SG ", security_groups)
        post_httprequest(url, security_groups)
+    print("SUCCESS: creating security groups")
     return id_list,device_ids
 
 
-def attach_subnets_to_router(rm_port, snm_port, router_id):
-    id_list, device_ids, ip_addrs = create_subnets_for_s4(snm_port)
+def attach_subnets_to_router(rm_port, snm_port, router_id, subnet_id_list):
     url = 'http://localhost:{}/project/{}/routers/{}/add_router_interface'.format(rm_port, get_projectid(),router_id)
-    print("attaching subnets to test cases")
+    print("Attaching subnets to router")
 
-    for id in id_list:
+    for id in subnet_id_list:
        subnet_info = {"subnet_id":id}
-       print("DDD: ASTR: subnet_info ", subnet_info)
-       print("DDD: ASTR: ARTI URL: ", url)
        put_httprequest(url, subnet_info)
     req="http://localhost:{}/project/{}/routers".format(rm_port, get_projectid())
-    print("ROUTER INFO")
+    print("Attached router info")
     print(get_httprequest(req))
-    return id_list,device_ids,ip_addrs
+    print("SUCCESS: attaching subnets to router")
 
+# Test case 1: L2 Basic
+# Two nodes in same subnet in different seurity groups
+def prepare_test_L2_basic(ip_mac, ser_port):
+    test_name = "L2_basic"
+    print("Preparing test case {}...".format(test_name))
+    serv = read_config_file_section("services")
+    create_default_segment_table(ser_port["vpm"])
+    create_vpc(ser_port["vpm"])
+    create_node(ser_port["nm"], ip_mac)
+    create_subnet(ser_port["snm"])
+    id_list,device_ids = create_security_groups(ser_port["sgm"])
+    change_ports = {"change":["security_groups","device_id"], "security_groups":id_list,"device_ids":device_ids}
+    create_port_goal_states(ser_port["pm"], change_ports)
+    ip_mac_db = get_mac_from_db()
+    print("Test case {}. IP/MAC in ignite DB: ".format(test_name, ip_mac_db))
+    print("SUCCESS: preparing test case {}...".format(test_name))
+    return ip_mac_db
 
-#test case 1 is with two nodes with different subnet ids in same same sg
-# S4
-def prepare_test_case_1(ip_mac, ser_port):
-    print("Test case 1 setup starts...")
+#test case 2 : L3 After Router (S4)
```

Comment:
It is a bit hard to tell what "L3 after router" and "L3 before router" mean. 

I would recommend the following name:
S4: AttachRouter_then_CreatePorts_L3
S5: CreatePorts_L3_then_AttachRouter